sources/engine/Stride.Navigation/Processors/NavigationProcessor.cs

```diff
@@ -89,10 +90,10 @@ protected override void OnEntityComponentRemoved(Entity entity, NavigationCompon
 
         private void DynamicNavigationMeshSystemOnNavigationMeshUpdatedUpdated(object sender, NavigationMeshUpdatedEventArgs eventArgs)
         {
-            var newNavigationMesh = eventArgs.BuildResult.NavigationMesh;
             NavigationMeshData data;
-            if (eventArgs.OldNavigationMesh != null && loadedNavigationMeshes.TryGetValue(eventArgs.OldNavigationMesh, out data))
+            if (eventArgs != null && eventArgs.OldNavigationMesh != null && loadedNavigationMeshes.TryGetValue(eventArgs.OldNavigationMesh, out data))
```

Comment:
Can be merged into a single check: `eventArgs?.OldNavigationMesh is not null`

Reply:
Thanks for point this out. Thinking on this again, I'm realizing `eventArgs` shouldn't be null anyway, and we need the old nav mesh to remove it from the processor when DynamicNavigationMeshSystem gets disabled. Have converted this pull to draft to fix this.

Reply:
Ok, so I decided to just null check `eventArgs` instead of ensuring it is never null unless advised otherwise.
`UpdateNavigationMesh` already accounts for unloading the stale mesh in the `UpdateNavigationMesh` method.
Also changing this behavior would technically be breaking change, though this would only occur when DynamicNavigationMeshSystem gets disabled (which was broken anyway).